src/clarity-angular/wizard/wizard-header-action.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2016-2017 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {
+    Component,
+    Input,
+    Output,
+    EventEmitter
+} from "@angular/core";
+
+let wizardHeaderActionIndex = 0;
+
+@Component({
+    selector: "clr-wizard-header-action",
+    template: `
+        <button class="btn clr-wizard-header-action btn-link"
+            [id]="id"
+            [class.disabled]="disabled"
+            (click)="click()"
+            [title]="title">
+            <ng-content></ng-content>
+        </button>
+    `,
+    host: {
+        "class": "clr-wizard-header-action-wrapper"
+    }
+})
+export class WizardHeaderAction {
+    // title is explanatory text added to the header action
+    @Input("title")
+    title: string = "";
+    // SPECME
+
+    // If our host has an ID attribute, we use this instead of our index.
+    @Input("id")
+    _id: string = (wizardHeaderActionIndex++).toString();
+
+    public get id(): string {
+        return `clr-wizard-header-action-${this._id}`;
+    }
+
+    @Input("clrWizardHeaderActionDisabled") public disabled: boolean = false;
+
+    @Output("actionClicked") headerActionClicked: EventEmitter <string> =
+        new EventEmitter(false);
+
+    click(): void {
+        if (this.disabled) {
+            return;
+        }
+
+        // passing the header action id allows users to have one method that
+        // routes to many different actions based on the type of header action
+        // clicked. this is further aided by users being able to specify ids
+        // for their header actions.
+        this.headerActionClicked.emit(this.id);
```

Comment:
I think you want to emit `this._id`, which is the id provided by the user. Otherwise, you return to them an id that you prefixed with a pretty long string, which seems strange.